packages/cli/src/commands/build/buildFrontend.ts

```diff
@@ -19,21 +19,60 @@ import { resolve as resolvePath } from 'path';
 import { buildBundle } from '../../lib/bundler';
 import { getEnvironmentParallelism } from '../../lib/parallel';
 import { loadCliConfig } from '../../lib/config';
+import chalk from 'chalk';
+import { BuildOptions } from '../../lib/bundler/types';
 
 interface BuildAppOptions {
   targetDir: string;
   writeStats: boolean;
   configPaths: string[];
+  isModuleFederationRemote?: true;
+}
+
+function getModuleFederationOptions(
+  name: string,
+  isRemote?: boolean,
+): BuildOptions['moduleFederation'] {
+  if (!isRemote && !process.env.EXPERIMENTAL_MODULE_FEDERATION) {
+    return undefined;
+  }
+
+  // The default output mode requires the name to be a usable as a code
+  // symbol, there might be better options here but for now we need to
+  // sanitize the name.
+  const sanitizedName = name.replaceAll('@', '').replaceAll(/[/\\_-]/g, '_');
+
+  console.log(
+    chalk.yellow(
+      `⚠️  WARNING: Module federation is experimental and will receive immediate breaking changes in the future.`,
+    ),
+  );
+
+  if (isRemote) {
+    return {
+      mode: 'remote',
+      name: sanitizedName,
+    };
+  }
+  return {
+    mode: 'host',
+    name: sanitizedName,
+  };
```

Comment:
Just a minor suggestion to shorten it up a bit

```suggestion
  return {
    mode: isRemote ? 'remote' : 'host',
    name: sanitizedName,
  };
```